tardis/model/base.py

```diff
@@ -317,17 +321,29 @@ def from_config(cls, config):
             t_inner = config.plasma.initial_t_inner
 
         abundances_section = config.model.abundances
+        isotope_index = pd.MultiIndex(
+            [[]] * 2, [[]] * 2, names=['atomic_number', 'mass_number'])
+        isotope_abundance = pd.DataFrame(columns=np.arange(no_of_shells),
+                                         index=isotope_index,
+                                         dtype=np.float64)
+
         if abundances_section.type == 'uniform':
             abundance = pd.DataFrame(columns=np.arange(no_of_shells),
                                      index=pd.Index(np.arange(1, 120),
                                                     name='atomic_number'),
                                      dtype=np.float64)
-
             for element_symbol_string in abundances_section:
                 if element_symbol_string == 'type':
                     continue
-                z = element_symbol2atomic_number(element_symbol_string)
-                abundance.ix[z] = float(abundances_section[element_symbol_string])
+                try:
```

Comment:
maybe this splitting - might be better performed earlier in the process and not in the init of the model (generally keeping logic out of inits is good practice).